# Activity 8

This activity tests your skills
in formulating an engineering problem as solving a system of linear algebraic equations,
as well as analyzing results thereof.

## Scenario

The lateral view of an [inclined strut](./act-08.png)
is designed as a ten-member truss.

You are tasked to analyze a scenario wherein
loads are approximated as concentrated forces 
acting on the non-support joints.
Members $AC$, $CE$, and $EG$ are collinear,
and perpendicular to the lines of action of loads $p_{1}$, $p_{2}$, and $p_{3}$.
Members $BD$, $DF$, and $FG$ are collinear,
and perpendicular
not just to the lines of action of loads $p_{4}$ and $p_{5}$,
but also to the supporting surface.

Let the dimensions, material makeup, and weight of the members be negligible.
Assume that the joints do not resist moments.
Determine the member forces developed
under such a conservative idealization of the strut.

## Modelling

Cast the original task as solving a system of linear algebraic equations
for the forces in $AC$, $BC$, $BD$, $CD$, $CE$, $DE$, $DF$, $EF$, $EG$, and $FG$.

Arrange the equations in the matrix-vector form
$\boldsymbol{K} \boldsymbol{\phi} = \boldsymbol{\delta}$,
where
$\boldsymbol{\phi}$ collects the member forces in the order presented above.
For uniformity, assume that tensile forces are positive.

## Core code

In [1]:
import math as mt

import numpy as np

Define a Python function
`get_coeffs_rhs()`
that takes four positional arguments
(namely, `d`, `h`, `theta`, `p1`, `p2`, `p3`, `p4`, `p5`,
in that order)
and returns two NumPy arrays
(namely,
`coeff_matrix`, `rhs_vector`
in that order).
The roles of these inputs and outputs should be evident.

In [2]:
def get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5):
    a = mt.atan(h/(3*d))
    b = mt.atan((2*h)/(3*d))
    t = theta*(mt.pi/180)
    y = (mt.pi/2)-t-a

    return np.array([
        [-mt.cos(y), -mt.cos(y+a+b), 0, mt.cos(t), mt.cos(y), 0, 0, 0, 0, 0],
        [-mt.sin(y), -mt.sin(y+a+b), 0, -mt.sin(t), mt.sin(y), 0, 0, 0, 0, 0],
        [0, 0, -mt.cos((mt.pi/2)-t), -mt.cos(t), 0, mt.cos((mt.pi/2)-t+a), mt.cos((mt.pi/2)-t), 0, 0, 0],
        [0, 0, -mt.sin((mt.pi/2)-t), mt.sin(t), 0, mt.sin((mt.pi/2)-t+a), mt.sin((mt.pi/2)-t), 0, 0, 0],
        [0, 0, 0, 0, -mt.cos(y), -mt.cos(y+(2*a)), 0, mt.cos(t), mt.cos(y), 0],
        [0, 0, 0, 0, -mt.sin(y), -mt.sin(y+(2*a)), 0, -mt.sin(t), mt.sin(y), 0],
        [0, 0, 0, 0, 0, 0, -mt.cos((mt.pi/2)-t), -mt.cos(t), 0, mt.cos((mt.pi/2)-t)],
        [0, 0, 0, 0, 0, 0, -mt.sin((mt.pi/2)-t), mt.sin(t), 0, mt.sin((mt.pi/2)-t)],
        [0, 0, 0, 0, 0, 0, 0, 0, -mt.cos(y), -mt.cos((mt.pi/2)-t)],
        [0, 0, 0, 0, 0, 0, 0, 0, -mt.sin(y), -mt.sin((mt.pi/2)-t)]         
    ]), np.array([
        -p1*mt.cos(t + a), 
        p1*mt.sin(t + a), 
        -p5*mt.cos(t), 
        p5*mt.sin(t), 
        -p2*mt.cos(t + a), 
        p2*mt.sin(t + a), 
        -p4*mt.cos(t),
        p4*mt.sin(t),
        -p3*mt.cos(t + a),
        p3*mt.sin(t + a)
    ])    

## Assessment

Define Python variables
`D`, `H`, `THETA`, `P1`, `P2`, `P3`, `P4`, and `P5`
for storing the values of parameters
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

Every time you solve for $\boldsymbol{\phi}$,
- use `get_coeffs_rhs()` to generate the coefficient matrix and the right-hand-side vector,
  respectively storing them in Python variables `K` and `d`;
- show the determinant and the condition number of the coefficient matrix;
  and
- print $\boldsymbol{\phi}$.


This activity uses 3 sets of values for the defined Python variables (as refered to the bottom of this file), which are assigned by group. In this case, we will use set 3, below are the values to be used:

In [3]:
D = 1
H = 6
THETA = 15
P1 = 1*(10**3)  
P2 = 9*(10**3)   
P3 = 8*(10**3)   
P4 = 7*(10**3)  
P5 = 5*(10**3)  

### Task 1

Determine the member forces when $C$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_1` to store $\boldsymbol{\phi}$ for this task.

In [4]:
K, d = get_coeffs_rhs(D, H, THETA, 1, 0, 0, 0, 0)

In [5]:
phi_1 = np.linalg.solve(K, d)

print(phi_1)

[-1.16666667 -1.53659074 -0.         -0.         -0.         -0.
 -0.         -0.         -0.         -0.        ]


In [6]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 2

Determine the member forces when $E$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_2` to store $\boldsymbol{\phi}$ for this task.

In [7]:
K, d = get_coeffs_rhs(D, H, THETA, 0, 1, 0, 0, 0)

In [8]:
phi_2 = np.linalg.solve(K, d)

print(phi_2)

[-0.33333333 -0.76829537 -0.55901699  1.11803399 -0.75       -1.25
 -0.         -0.         -0.         -0.        ]


In [9]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 3

Determine the member forces when $G$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_3` to store $\boldsymbol{\phi}$ for this task.

In [10]:
K, d = get_coeffs_rhs(D, H, THETA, 0, 0, 1, 0, 0)

In [11]:
phi_3 = np.linalg.solve(K, d)

print(phi_3)

[ 5.00000000e-01  9.31309157e-17 -1.11803399e+00 -1.05123726e-16
  5.00000000e-01  5.05073305e-17 -1.11803399e+00 -5.99485963e-17
  5.00000000e-01 -1.11803399e+00]


In [12]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 4

Determine the member forces when $F$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_4` to store $\boldsymbol{\phi}$ for this task.

In [13]:
K, d = get_coeffs_rhs(D, H, THETA, 0, 0, 0, 1, 0)

In [14]:
phi_4 = np.linalg.solve(K,d)

print(phi_4)

[ 0.74535599 -0.34359214 -0.25        0.5         0.55901699 -0.55901699
 -0.          1.         -0.         -0.        ]


In [15]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 5

Determine the member forces when $D$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_5` to store $\boldsymbol{\phi}$ for this task.

In [16]:
K, d = get_coeffs_rhs(D, H, THETA, 0, 0, 0, 0, 1)

In [17]:
phi_5 = np.linalg.solve(K, d)

print(phi_5)

[ 0.372678   -0.68718427 -0.          1.         -0.         -0.
 -0.         -0.         -0.         -0.        ]


In [18]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 6

Determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces.

Define a Python variable `phi_6` to store $\boldsymbol{\phi}$ for this task.

In [19]:
K, d = get_coeffs_rhs(D, H, THETA, 1, 1, 1, 1, 1)

In [20]:
phi_6 = np.linalg.solve(K, d)

print(phi_6)

[ 0.11803399 -3.33566252 -1.92705098  2.61803399  0.30901699 -1.80901699
 -1.11803399  1.          0.5        -1.11803399]


In [21]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 7

Similar to Task 6,
determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces,
but use the results in Tasks 1 - 5.

Define a Python variable `phi_7` to store $\boldsymbol{\phi}$ for this task.

In [22]:
phi_7 = phi_1 + phi_2 + phi_3 + phi_4 + phi_5

print(phi_7)

[ 0.11803399 -3.33566252 -1.92705098  2.61803399  0.30901699 -1.80901699
 -1.11803399  1.          0.5        -1.11803399]


### Task 8

Determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define a Python variable `phi_8` to store $\boldsymbol{\phi}$ for this task.

In [23]:
K, d = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

In [24]:
phi_8 = np.linalg.solve(K, d)

print(phi_8)

[  6914.21526208 -14292.31538881 -15725.42485937  18562.30589875
   1163.11896062 -15163.11896062  -8944.27191      7000.
   4000.          -8944.27191   ]


In [25]:
det = np.linalg.det(K)
cond = np.linalg.cond(K)
print(f"Determinant: {det}")
print("")
print(f"Condition Number: {cond}")

Determinant: 0.46566840006975907

Condition Number: 6.542591657117212


### Task 9

Similar to Task 8,
determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
but use the results in Tasks 1 - 5.


Define a Python variable `phi_9` to store $\boldsymbol{\phi}$ for this task.

In [26]:
phi_9 = phi_1*P1 + phi_2*P2 + phi_3*P3 + phi_4*P4 + phi_5*P5

print(phi_9)

[  6914.21526208 -14292.31538881 -15725.42485937  18562.30589875
   1163.11896062 -15163.11896062  -8944.27191      7000.
   4000.          -8944.27191   ]


### Task 10

Determine the magnitudes of the support reactions
using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define Python variables `rxn_A` and `rxn_B` 
to store the forces developed at $A$ and $B$, respectively.

In [27]:
rxn_A = mt.sqrt((-phi_9[0])**2)

b = mt.atan((2*H)/(3*D))

Bx = phi_9[2] + phi_9[1]*mt.cos(b)
By = phi_9[1]*mt.sin(b)
rxn_B = mt.sqrt(Bx**2 + By**2)

print(f"Magnitude of the reaction at A is:{rxn_A}")
print(f"Magnitude of the reaction at B is:{rxn_B}")

Magnitude of the reaction at A is:6914.215262082665
Magnitude of the reaction at B is:23676.578188378455


## Instructions

Do not use any library or module other than those in the Imports section.

### Parameters

For each set,
the values shown are that of
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

- Set 1: 1.5 ft, 4.5 ft, 21°, 5 kN, 7 kN, 6 kN, 8 kN, 1 kN
- Set 2: 2.0 ft, 3.0 ft, 17°, 9 kN, 6 kN, 5 kN, 1 kN, 7 kN
- Set 3: 1.0 ft, 6.0 ft, 15°, 1 kN, 9 kN, 8 kN, 7 kN, 5 kN

### Scoring

In each of Tasks 1 - 9,
obtaining a correct member force merits one (1) point.

In each Task but 7, 9, and 10,
- obtaining a correct determinant merits one (1) point,
  and
- obtaining a correct condition number merits one (1) point.

For Task 10,
obtaining a correct reaction force merits five (5) points.

Every violation of an instruction
means a deduction of one (1) point.

All in all, one may earn up to 124 points for this activity.

### Submission

Download this notebook file,
and save with a filename following the pattern
`ACT-08_<Group name>`,
where the group name is as listed in class.
For example, if you belong to the group SixIsEven,
then your notebook should be named `ACT-08_SixIsEven.ipynb`.
Submit your notebook via the classwork platform for this activity in Google Classroom.
Submissions beyond the deadline will not be considered.

Lastly, the use of AI tools to answer this exam is not prohibited,
but it is of ethical interest to disclose such use.
This is in line with the
[MSU Policy on the Fair and Ethical Use of AI and Its Applications](https://www.msumain.edu.ph/wp-content/uploads/2024/05/MSU-Policy-on-Ethical-use-of-AI-Policies.pdf).
As such, please include a brief statement (in a private comment to this classwork)
declaring which and how AI tools are used in your work.

*Last updated by Christian Cahig on 2025-11-25*